app/src/main/java/com/mapbox/services/android/navigation/testapp/activity/navigationui/WaypointNavigationActivity.java

```diff
@@ -190,8 +192,11 @@ private void fetchRoute(Point origin, Point destination) {
       .build()
       .getRoute(new SimplifiedCallback() {
         @Override
-        public void onResponse(Call<DirectionsResponse> call, Response<DirectionsResponse> response) {
-          startNavigation(response.body().routes().get(0));
+        public void onResponse(@NotNull Call<DirectionsResponse> call, @NotNull Response<DirectionsResponse> response) {
+          DirectionsResponse directionsResponse = response.body();
+          if (directionsResponse != null) {
+            startNavigation(directionsResponse.routes().get(0));
```

Comment:
wouldn't `get(0)` call on an empty list result in `IndexOutOfBoundsException`?

Reply:
Done